BinarySerializer/Graph/ValueGraph/ValueNode.cs

```diff
@@ -201,9 +201,17 @@ public virtual void BindChecks()
 
                 if (offset != null)
                 {
-                    using (new StreamResetter(stream))
+                    if (TypeNode.OffsetRewind)
                     {
-                        stream.Position = offset.Value;
+                        using (new StreamResetter(stream))
+                        {
+                            stream.Seek(offset.Value, TypeNode.OffsetSeekOrigin);
+                            SerializeInternal(stream, GetConstFieldLength, eventShuttle, measuring);
+                        }
+                    }
+                    else
+                    {
+                        stream.Seek(offset.Value, TypeNode.OffsetSeekOrigin);
                         SerializeInternal(stream, GetConstFieldLength, eventShuttle, measuring);
```

Comment:
Could this whole thing be condensed to something like?
```C#
var rewindPosition = stream.Position;
stream.Seek(offset.Value, TypeNode.OffsetSeekOrigin);
SerializeInternal(stream, GetConstFieldLength, eventShuttle, measuring);
if (TypeNode.OffsetRewind)
    stream.Position = rewindPosition;
```

Unless the StreamResetter does more than I'm thinking it does.  At this stage I imagine it just captures the current position when instantiated, and the Dispose restores the captured position.

Reply:
Good cath, I've updated